### PR TITLE
sudo: pass env variables without /usr/bin/env

### PR DESCRIPTION
**Pull request by Kentzo**
Using /usr/bin/env as a frontend for the actual command prevents sudoers from restricting allowed commands and configuring detailed command environment.

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by Bo98** [comment]
Looks like this might break the case where envs are used but sudo is _not_ used?

File: Library/Homebrew/system_command.rb [hunk 1]
```diff
@@ -154,15 +154,23 @@ def env_args
...
+
+  sig { returns(T::Array[String]) }
+  def env_previx
+    ["/usr/bin/env", *env_args]
+  end
+
+  sig { returns(T::Array[String]) }
+  def command_prefix
+    sudo? ? sudo_prefix : env_previx
   end
```
Comment (MikeMcQuaid) [inline Library/Homebrew/system_command.rb:174]:
```suggestion
  def env_prefix
    ["/usr/bin/env", *env_args]
  end

  sig { returns(T::Array[String]) }
  def command_prefix
    sudo? ? sudo_prefix : env_prefix
  end
```

**Comment by MikeMcQuaid** [comment]
@Bo98 can you elaborate a bit with an example case?

**Comment by Bo98** [comment]
My concerns were addressed in a later force push.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @Kentzo!

**Pull request merged**